### PR TITLE
[stable10] catch duplicate token inserts and log changes

### PR DESCRIPTION
**Pull request by butonic**
there are codepaths that will cause the session token to be generated twice.
This PR can be used to debug the token related changes.

use conditional logging to create a separate log file:
```php
  'log.conditions' => [
		[
			'apps' => [
				'OC\\User\\Session::login',
				'OC\\User\\Session::loginWithApache',
				'OC\\User\\Session::loginWithCookie',
				'OC\\User\\Session::createSessionToken',
				'OC\\User\\Session::checkTokenCredentials',
				'OC\\User\\Session::validateToken',
				'OC\\User\\Session::checkTokenCredentials',
				'OC\\Authentication\\Token\\DefaultTokenProvider::generateToken',
				'OC\\Authentication\\Token\\DefaultTokenProvider::getToken',
				'OC\\Authentication\\Token\\DefaultTokenProvider::updateToken',
				'OC\\Authentication\\Token\\DefaultTokenProvider::updateTokenActivity',
				'OC\\Authentication\\Token\\DefaultTokenProvider::invalidateToken',
				'OC\\Authentication\\Token\\DefaultTokenProvider::invalidateTokenById',
				'OC\\Authentication\\Token\\DefaultTokenProvider::invalidateOldTokens',
			],
			'logfile' => '/tmp/oc-auth-token.log'
		],
  ],
```

cc @tomneedham

### CODE REVIEW

**Comment by butonic** [comment]
forward port to master required

**Review by PVince81 — APPROVED** [review]
Code looks fine, as discussed.

Should only require smoke testing that login still works.

**Comment by phil-davis** [comment]
Rebase after https://github.com/owncloud/core/pull/31762 is merged to ``stable10`` and the drone fail will go away.

**Comment by PVince81** [comment]
master: https://github.com/owncloud/core/pull/31778

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -701,7 +719,17 @@ public function createSessionToken(IRequest $request, $uid, $loginName, $passwor
 		} catch (SessionNotAvailableException $ex) {
 			// This can happen with OCC, where a memory session is used
 			// if a memory session is used, we shouldn't create a session token anyway
+			$this->logger->logException($ex, ['app' => __METHOD__]);
```
Comment (patrickjahns) [inline lib/private/User/Session.php]: I fear that this will create havoc / bogus entries

**Pull request merged**